File: Cpp/1854_Maximum_Population_Year.cpp
class Solution {
public:
    int maximumPopulation(vector<vector<int>>& logs) {
        int result = 0;
        
        vector<int> population(2051, 0);
        
        for(auto& log : logs) {
            population[log[0]] += 1;
            population[log[1]] -= 1;
        }
        
        int max_population = 0;
        int current = 0;
        for(int i = 1950; i <= 2050; ++i) {
            current += population[i];
            
            if(current > max_population){
                max_population = current;
                result = i;
            }
        }
        
        return result;
    }
};

File: Kotlin/1854_Maximum_Population_Year.kt
class Solution {
    fun maximumPopulation(logs: Array<IntArray>): Int {
        var result = 0
        val population = IntArray(2051)
        
        for(log in logs) {
            ++population[log[0]]
            --population[log[1]]
        }
        
        var maxPopulation = 0
        var current = 0
        for(i in 1950..2050) {
            current += population[i]

            if(current > maxPopulation){
                maxPopulation = current
                result = i
            }
        }
        
        return result
    }
}